src/plugins/data/common/es_query/kuery/node_types/types.ts

```diff
@@ -58,17 +58,17 @@ interface LiteralType {
   toElasticsearchQuery: (node: any) => null | boolean | number | string;
 }
 
-interface LiteralTypeBuildNode {
+export interface LiteralTypeBuildNode {
   type: 'literal';
   value: null | boolean | number | string;
 }
 
 interface NamedArgType {
   buildNode: (name: string, value: any) => NamedArgTypeBuildNode;
-  toElasticsearchQuery: (node: any) => string;
+  toElasticsearchQuery: (node: any) => JsonValue;
```

Comment:
For each of the `toElasticsearchQuery()` items in this file, should the `node` param they are accepting match the returned value of the `buildNode()` function?

For example here, I would expect the interface to be:
```ts
interface NamedArgType {
  buildNode: (name: string, value: any) => NamedArgTypeBuildNode;
  toElasticsearchQuery: (node: NamedArgTypeBuildNode) => JsonValue;
}
```

...and similar for each of the others as well as the individual node_types definitions themselves.

I don't know enough about the kuery implementation to say with 100% confidence that this should be the case, but it makes sense based on my understanding of the code and would make the typings in this PR much simpler

Reply:
I agree with you, but I was trying to stick with the API defined in `types.ts` as much as possible, to avoid errors.